### PR TITLE
[WIP] single base class for requests and handles

### PR DESCRIPTION
**Pull request by piscisaureus**
Continuation of #102

### CODE REVIEW

File: include/uv.h [hunk 1]
```diff
@@ -1396,21 +1391,9 @@ UV_EXTERN uv_thread_t uv_thread_self(void);
 UV_EXTERN int uv_thread_join(uv_thread_t *tid);
 UV_EXTERN int uv_thread_equal(const uv_thread_t* t1, const uv_thread_t* t2);
 
-/* The presence of these unions force similar struct layout. */
```
Comment (saghul) [inline include/uv.h]: Don't we need anything similar to this?

**Comment by saghul** [comment]
Generally it looks good. I wonder if we should still have some separation between handles and requests, like `uv_any_handle` and `uv_any_req`, even if both handles and requests are `uv_object_t` underneath.

**Comment by saghul** [comment]
@piscisaureus, when you come back to this, I think a (short) LEP would be in order :-)

**Comment by cjihrig** [comment]
I'm going to close this, as it's been 2 years and needs to be rebased. Feel free to reopen if you get back to this.

**Pull request closed without merging**